src/web/common.rkt

```diff
@@ -67,44 +67,90 @@
                               ,@(for/list ([line extra])
                                   `(li ,line)))))))))
 
+(define (render-comparison #:title [title #f] name a b )
+  (render-large #:title title name a `(span ([class "unit"]) " → ") b))
+
 (define (render-large #:title [title #f] name . values)
   `(div ,name ": " (span ([class "number"]
                           ,@(if title `([title ,title]) '()))
                          ,@values)))
 
-(define (render-comparison #:title [title #f] name a b )
-  (render-large #:title title name a `(span ([class "unit"]) " → ") b))
+(define (preprocess->c-like preprocess #:assert-has-parentheses [assert-has-parentheses #t])
+  (match preprocess
+    [(list 'sort variables ...)
+     (format
+      (if assert-has-parentheses
+          "assert(~a);"
+          "assert ~a;")
+      (string-join
+       (for/list ([a (in-list variables)]
+                  [b (in-list (cdr variables))])
+         (format "~a < ~a" a b))
+       " && "))]))
+
+(define (preprocess->c preprocess)
+  (preprocess->c-like preprocess #:assert-has-parentheses #t))
+
+(define (preprocess->java preprocess)
+  (preprocess->c-like preprocess #:assert-has-parentheses #f))
+
+(define (preprocess->python preprocess)
+  (match preprocess
+    [(list 'sort variables ...)
+     (define lst (format "[~a]" (string-join (map ~a variables) ", ")))
+     (format "~a = sort(~a)" lst lst)]))
+
+(define (preprocess->julia preprocess)
+  (match preprocess
+    [(list 'sort variables ...)
+      (define list-without-brackets (string-join (map ~a variables) ", "))
+      (define list-with-brackets (format "[~a]" list-without-brackets))
+      (format "~a = sort(~a)" list-without-brackets list-with-brackets)]))
```

Comment:
Just add the brackets in this `format` call, and then you won't need two variants. The same applies in lots of other preprocess functions.

Reply:
Yeah this looks quite a bit cleaner